test/tests/functional/pbs_cray_check_node_exclusivity.py

```diff
@@ -0,0 +1,560 @@
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, offset=10)
+        self.server.expect(NODE, {'state': 'resv-exclusive'},
+                           id=resv_node)
+        # Delete future reservation rid2 and verify that resv node
+        # is still in state resv-exclusive
+        self.server.delete(rid2)
+        self.server.expect(NODE, {'state': 'resv-exclusive'},
+                           id=resv_node)
+
+    def test_job_inside_exclusive_reservation(self):
+        """
+        Test Job will run correctly inside the exclusive
+        reservation
+        """
+        self.script2 = []
+        self.script2 += ['echo Hello World\n']
+        self.script2 += ['/bin/sleep 10']
+
+        # Submit a reservation with place=excl
+        now = int(time.time())
+        a = {'Resource_List.select': '1:ncpus=1:vntype=cray_login',
+             'Resource_List.place': 'excl', 'reserve_start': now + 20,
+             'reserve_end': now + 40}
+        r = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(r)
+        rid_q = rid.split('.')[0]
+        a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, a, id=rid)
+        node = self.server.status(RESV, 'resv_nodes', id=rid)
+        resv_node = self.server.reservations[rid].get_vnodes()[0]
+        self.server.expect(NODE, {'state': 'free'},
+                           id=resv_node)
+        self.logger.info('Waiting 20s for reservation to start')
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, offset=20)
+        self.server.expect(NODE, {'state': 'resv-exclusive'},
+                           id=resv_node)
+        # Submit a jinside the reservation
+        submit_dir = self.du.create_temp_dir(asuser=TEST_USER)
+        a = {ATTR_q: rid_q, ATTR_l + '.select': '1:ncpus=1:vntype=cray_login',
+             'Resource_List.place': 'excl'}
+        j1 = Job(TEST_USER, attrs=a)
+        j1.create_script(self.script2)
+        jid1 = self.server.submit(j1, submit_dir=submit_dir)
+        self.server.expect(
+            JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(NODE, {'state': 'job-exclusive,resv-exclusive'},
+                           id=resv_node)
+        # waith 5 sec for job to end
+        self.server.expect(NODE, {'state': 'resv-exclusive'},
+                           id=resv_node, offset=5, interval=10)
+        # Wait for reservation to end and verify node state
+        # changed as free
+        msg = "Que;" + rid_q + ";deleted at request of pbs_server@"
+        self.server.log_match(msg, starttime=now, interval=2)
+        self.server.expect(NODE, {'state': 'free'},
+                           id=resv_node)
+
+    def test_job_inside_exclusive_reservation_on_compute_node(self):
+        """
+        Test Job will run correctly inside the exclusive
+        reservation requesting compute_node
```

Comment:
Combine this test with the above test.  Just submit the cray compute job, check for success and then do it again for the cray login job.